games-emulation/retroarch/retroarch-1.7.1.recipe

```diff
@@ -0,0 +1,87 @@
+SUMMARY="The reference frontend of libretro"
+DESCRIPTION="RetroArch is a frontend for emulators, game engines and media \
+players. It enables you to run classic games on a wide range of computers and \
+consoles through its slick graphical interface. It has advanced features like \
+shaders, netplay, rewinding, next-frame response times, and more!"
+HOMEPAGE="https://libretro.com/"
+COPYRIGHT="2010-2018 The RetroArch Team"
+LICENSE="GNU GPL v3"
+REVISION="1"
+SOURCE_URI="https://github.com/libretro/RetroArch/archive/v$portVersion.tar.gz"
+CHECKSUM_SHA256="e9189c7da6b2ca0142528eca14f26104ee322f052f8609203d90b60938d8d3f3"
+SOURCE_FILENAME="retroarch-$portVersion.tar.gz"
+SOURCE_DIR="RetroArch-$portVersion"
+
+srcGitRev_2="505ad6205e5fd8a155daad02ab39953a0b15b0a9"
+SOURCE_URI_2="https://github.com/libretro/retroarch-assets/archive/$srcGitRev_2.tar.gz"
+CHECKSUM_SHA256_2="de37772b00b755b80aafe5d4df0713c0fe6b0e5a79017d94e43f279e798aa85d"
+SOURCE_FILENAME_2="retroarch-assets-$portVersion.tar.gz"
+
+ARCHITECTURES="!x86_gcc2 ?x86 x86_64"
+SECONDARY_ARCHITECTURES="x86"
+
+PROVIDES="
+	retroarch$secondaryArchSuffix = $portVersion
+	app:retroarch$secondaryArchSuffix = $portVersion
+	cmd:retroarch$secondaryArchSuffix = $portVersion
+	cmd:retroarch_cg2glsl$secondaryArchSuffix = $portVersion
+	"
+REQUIRES="
+	haiku$secondaryArchSuffix
+	lib:libsdl2_2.0$secondaryArchSuffix
+	lib:libGL$secondaryArchSuffix
+	lib:libz$secondaryArchSuffix
+	lib:libavcodec$secondaryArchSuffix
+	lib:libavformat$secondaryArchSuffix
+	lib:libavdevice$secondaryArchSuffix
+	lib:libswresample$secondaryArchSuffix
+	lib:libavresample$secondaryArchSuffix
+	lib:libavutil$secondaryArchSuffix
+	lib:libswscale$secondaryArchSuffix
+	"
+
+BUILD_REQUIRES="
+	haiku${secondaryArchSuffix}_devel
+	devel:libsdl2$secondaryArchSuffix
+	devel:libgl$secondaryArchSuffix
+	devel:libz$secondaryArchSuffix
+	"
+BUILD_PREREQUIRES="
+        cmd:gcc$secondaryArchSuffix
+        cmd:git
+	cmd:make
+        cmd:pkg_config$secondaryArchSuffix
+	cmd:which
+	"
+GLOBAL_WRITABLE_FILES="
+	$relativeSettingsDir/retroarch.cfg keep-old
+"
```

Comment:
I'm getting this upon package install:
````
[system] warning: package retroarch-1.7.1-1-x86_64.hpkg: "/boot/system/settings/retroarch.cfg" cannot be updated automatically, since it doesn't have a SYS:PACKAGE attribute. Please perform the update manually if needed.
````
Is it normal? It's just a warning so I don't worry much, but just want to make sure it's fine.

Reply:
I guess it's not normal, but I'm not sure because I never got this message.
It might as well be a consequence of an incorrect settings for `GLOBAL_WRITABLE_FILES` and/or `USER_SETTINGS_FILES`.

Did you install the hpkg with `pkgman install retroarch-1.7.1-1-x86_64.hpkg` or did you copy the hpkg to `/system/packages/` ?
Both work, but I prefer using pkgman (instead of cp) because when the installation requires older versions to be uninstalled or when there are conflicts, pkgman asks the user what to do on the console, whereas if you install by copying the hpkg, the questions are displayed in the GUI.

Also, you cannot re-install the exact same version&revision of a package if it is already installed.
If there is no revision bump, then you need to uninstall the previous hpkg (with `pkgman uninstall retroarch`) and then re-install your new hpkg.

A workaround that I use for packages which I cannot uninstall (without having to accept to uninstall all the packages that depend on it) is that I keep a copy of the previous hpkg. So I first install an old hpkg and then I install the new one.
(BTW, there is currently no way (to my knowledge) to have pkgman replace a package by a new one with the same version and revision. We can cheat with `touch -r` and `cp -p` but after this we need to reboot.)

Reply:
I installed it with `pkgman` and I believe I have been getting this message because `retroarch` was already installed.
Good to know however that `pkgman` won't overwrite an already existing package with the same revision.
I just tried again and here goes:
````
~> pkgman uninstall retroarch
The following changes will be made:
  in system:
    uninstall package retroarch-1.7.1-3
Continue? [yes/no] (yes) : yes[system] Applying changes ...
[system] Changes applied. Old activation state backed up in "state_2018-02-27_16:05:50"
[system] Cleaning up ...
[system] Done.
~> pkgman install /boot/home/haikuports/packages/retroarch-1.7.1-1-x86_64.hpkg
100% repochecksum-1 [65 bytes]
Validating checksum for Haiku...done.
100% repocache-2 [3.65 KiB]
Validating checksum for Haiku...done.
100% repochecksum-1 [64 bytes]
Validating checksum for HaikuPorts...done.
100% repocache-2 [676.07 KiB]
Validating checksum for HaikuPorts...done.
The following changes will be made:
  in system:
    install package retroarch-1.7.1-1 from local file
Continue? [yes/no] (yes) : [system] Applying changes ...
[system] Changes applied. Old activation state backed up in "state_2018-02-27_16:06:21"
[system] Cleaning up ...
[system] Done.
````
Looks like we're good.
Again I would like to reiterate my thanks for all your patient help :)